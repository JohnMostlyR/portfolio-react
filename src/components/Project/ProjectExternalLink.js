import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import typography from '../../styles/templates/typography';
import {visuallyHidden} from '../../styles/mixins';

const ListItem = styled.li`
  display: inline-block;
  margin-right: 1vw;
  transition: transform .2s ease-in-out;
  vertical-align: middle;

  &:hover {
    transform: scale(1.5);
  }
`;

const Link = styled.a`
  color: #fff;
  ${typography.doublePica}
`;

const LinkDescription = styled.span`
  ${visuallyHidden}
`;

const ProjectExternalLink = (props) => (
  <ListItem data-tip={props.name}>
    <Link
      href={props.url}
      target="_blank"
      rel="noopener noreferrer">
      <span className={props.icon} aria-hidden="true"/>
      <LinkDescription>{props.name}</LinkDescription>
    </Link>
  </ListItem>
);

ProjectExternalLink.propTypes = {
  icon: PropTypes.string,
  name: PropTypes.string,
  url: PropTypes.string,
};

ProjectExternalLink.defaultPropTypes = {
  name: 'Name',
  url: '#',
};

export default ProjectExternalLink;
