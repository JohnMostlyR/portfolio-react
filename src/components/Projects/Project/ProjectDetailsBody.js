import styled from 'styled-components';

const ProjectDetailsBody = styled.p`
  min-width: 32vw;
  margin-top: .5em;
`;

export default ProjectDetailsBody;