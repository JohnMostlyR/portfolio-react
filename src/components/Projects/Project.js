import React from 'react';
import styled from 'styled-components';
import mq from '../../styles/templates/mediaQueries';
import typography from '../../styles/templates/typography';
import ProjectExternalLink from './ProjectExternalLink';
import MediaAsset from '../../components/MediaAsset';

const Article = styled.article`
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
`;

const Body = styled.div`
  flex: 1 0 50%;
  order: 2;
  padding-bottom: 2vw;
  padding-left: 2.5vw;
  padding-right: 2.5vw;
  padding-top: 3vw;
  
  ${mq.m`
    order: 1;
    padding-bottom: 1vw;
    padding-left: 3vw;
    padding-top: 1vw;
  `}
`;

const TitleWrapper = styled.div`
  display: flex;
  width: 100%;
`;

const Title = styled.h2`
  flex: 1;
  font-family: Consolas, Monaco, "Andale Mono", monospace !important;
  hyphens: auto;
  overflow-wrap: break-word;
  word-wrap: break-word;
  -ms-word-break: break-all;
  word-break: break-word;
  
  ${typography.doublePica}
`;

const ProjectLinkslist  = styled.ul`
  height: 1.5em;
  margin: 0;
  padding: 0;
  list-style: none;
  text-align: right;
`;

const DetailsTitle = styled.p`
  margin-top: .25em;
  ${typography.pica}
`;

const DetailsBodyText = styled.p`
  min-width: 32vw;
  margin-top: .5em;
`;

const Project = (props) => {
  return (
    <Article>
      <MediaAsset imageSource={props.thumbnailUrl} isOdd={props.isOdd}/>
      <Body>
        <div>
          <TitleWrapper>
            <Title>{props.title}</Title>
            <ProjectLinkslist>
              {
                props.links.map(link => <ProjectExternalLink key={link.name} link={link}/>)
              }
            </ProjectLinkslist>
          </TitleWrapper>
          <DetailsTitle><strong>{props.detailsTitle}</strong></DetailsTitle>
        </div>
        <DetailsBodyText>{props.detailsBodyText}</DetailsBodyText>
      </Body>
    </Article>
  );
};

export default Project;
