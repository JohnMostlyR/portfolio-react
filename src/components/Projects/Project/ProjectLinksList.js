import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import ProjectExternalLink from './ProjectExternalLink';

const UL = styled.ul`
  height: 1.5em;
  margin: 0;
  padding: 0;
  list-style: none;
  text-align: right;
`;

const ProjectLinksList = (props) => (
  <UL>
    {
      props.links.map(
        (link) => (
          <ProjectExternalLink
            key={link.name}
            name={link.name}
            url={link.url}
            icon={link.icon}
          />
        )
      )
    }
  </UL>
);

ProjectLinksList.propTypes = {
  title: PropTypes.string,
  links: PropTypes.array,
};

export default ProjectLinksList;