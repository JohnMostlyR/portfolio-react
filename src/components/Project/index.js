import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import ReactTooltip from 'react-tooltip';

import mq from '../../styles/templates/mediaQueries';
import typography from '../../styles/templates/typography';
import MediaAsset from '../MediaAsset';

import ProjectLinksList from './ProjectLinksList';

const Article = styled.article`
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
`;

const ProjectBody = styled.div`
  flex: 1 0 50%;
  order: 2;
  padding-bottom: 2vw;
  padding-left: 2.5vw;
  padding-right: 2.5vw;
  padding-top: 3vw;
  
  ${mq.m`
    order: 1;
    padding-bottom: 1vw;
    padding-left: 3vw;
    padding-top: 1vw;
  `}
`;

const ProjectHeader = styled.div`
  display: flex;
  width: 100%;
`;

const ProjectTitle = styled.h2`
  flex: 1;
  font-family: Consolas, Monaco, "Andale Mono", monospace !important;
  hyphens: auto;
  overflow-wrap: break-word;
  word-wrap: break-word;
  -ms-word-break: break-all;
  word-break: break-word;
  
  ${typography.doublePica}
`;

const ProjectDetailsTitle = styled.p`
  margin-top: .25em;
  ${typography.pica}
`;

const ProjectDetailsBody = styled.p`
  min-width: 32vw;
  margin-top: .5em;
  ${typography.bodyCopy};
`;

const Project = (props) => (
    <Article>
      <MediaAsset imageSource={props.thumbnailUrl} isOdd={props.isOdd}/>
      <ProjectBody>
        <div>
          <ProjectHeader>
            <ProjectTitle>{props.title}</ProjectTitle>
            <ProjectLinksList links={props.links}/>
          </ProjectHeader>
          <ProjectDetailsTitle><strong>{props.detailsTitle}</strong></ProjectDetailsTitle>
        </div>
        <ProjectDetailsBody>{props.detailsBodyText}</ProjectDetailsBody>
      </ProjectBody>
      <ReactTooltip />
    </Article>
);

Project.propTypes = {
  detailsTitle: PropTypes.string,
  detailsBodyText: PropTypes.string,
  isOdd: PropTypes.bool,
  links: PropTypes.array,
  thumbnailUrl: PropTypes.string,
  title: PropTypes.string,
};

Project.defaultProps = {
  title: 'New Project',
};

export default Project;
