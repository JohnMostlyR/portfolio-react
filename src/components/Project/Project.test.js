import React from 'react';
import {shallow} from 'enzyme';

import Project from './index';

describe('Project', () => {
  const links = [
    {
      name: 'github',
      url: 'https://github.com/',
      faIcon: 'github',
    },
    {
      name: 'codepen',
      url: 'https://codepen.io/',
      faIcon: 'codepen',
    },
  ];

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Project links={links}/>);
  });

  it('should render and match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});