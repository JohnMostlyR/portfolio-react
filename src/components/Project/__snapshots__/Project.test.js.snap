// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project should render and match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Project
    links={
        Array [
            Object {
              "faIcon": "github",
              "name": "github",
              "url": "https://github.com/",
            },
            Object {
              "faIcon": "codepen",
              "name": "codepen",
              "url": "https://codepen.io/",
            },
          ]
    }
    title="New Project"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <MediaAsset
          imageSource={undefined}
          isOdd={undefined}
/>,
        <styled.div>
          <div>
                    <styled.div>
                              <styled.h2>
                                        New Project
                              </styled.h2>
                              <ProjectLinksList
                                        links={
                                                  Array [
                                                            Object {
                                                              "faIcon": "github",
                                                              "name": "github",
                                                              "url": "https://github.com/",
                                                            },
                                                            Object {
                                                              "faIcon": "codepen",
                                                              "name": "codepen",
                                                              "url": "https://codepen.io/",
                                                            },
                                                          ]
                                        }
                              />
                    </styled.div>
                    <styled.p>
                              <strong />
                    </styled.p>
          </div>
          <styled.p />
</styled.div>,
        <ReactTooltip
          insecure={true}
          resizeHide={true}
          wrapper="div"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "imageSource": undefined,
          "isOdd": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <div>
              <styled.div>
                            <styled.h2>
                                          New Project
                            </styled.h2>
                            <ProjectLinksList
                                          links={
                                                        Array [
                                                                      Object {
                                                                        "faIcon": "github",
                                                                        "name": "github",
                                                                        "url": "https://github.com/",
                                                                      },
                                                                      Object {
                                                                        "faIcon": "codepen",
                                                                        "name": "codepen",
                                                                        "url": "https://codepen.io/",
                                                                      },
                                                                    ]
                                          }
                            />
              </styled.div>
              <styled.p>
                            <strong />
              </styled.p>
</div>,
            <styled.p />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <styled.div>
                  <styled.h2>
                                    New Project
                  </styled.h2>
                  <ProjectLinksList
                                    links={
                                                      Array [
                                                                        Object {
                                                                          "faIcon": "github",
                                                                          "name": "github",
                                                                          "url": "https://github.com/",
                                                                        },
                                                                        Object {
                                                                          "faIcon": "codepen",
                                                                          "name": "codepen",
                                                                          "url": "https://codepen.io/",
                                                                        },
                                                                      ]
                                    }
                  />
</styled.div>,
                <styled.p>
                  <strong />
</styled.p>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <styled.h2>
                      New Project
</styled.h2>,
                    <ProjectLinksList
                      links={
                                            Array [
                                                                  Object {
                                                                    "faIcon": "github",
                                                                    "name": "github",
                                                                    "url": "https://github.com/",
                                                                  },
                                                                  Object {
                                                                    "faIcon": "codepen",
                                                                    "name": "codepen",
                                                                    "url": "https://codepen.io/",
                                                                  },
                                                                ]
                      }
/>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "class",
                    "props": Object {
                      "children": "New Project",
                    },
                    "ref": null,
                    "rendered": "New Project",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "function",
                    "props": Object {
                      "links": Array [
                        Object {
                          "faIcon": "github",
                          "name": "github",
                          "url": "https://github.com/",
                        },
                        Object {
                          "faIcon": "codepen",
                          "name": "codepen",
                          "url": "https://codepen.io/",
                        },
                      ],
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": <strong />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "strong",
                },
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "insecure": true,
          "resizeHide": true,
          "wrapper": "div",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <MediaAsset
            imageSource={undefined}
            isOdd={undefined}
/>,
          <styled.div>
            <div>
                        <styled.div>
                                    <styled.h2>
                                                New Project
                                    </styled.h2>
                                    <ProjectLinksList
                                                links={
                                                            Array [
                                                                        Object {
                                                                          "faIcon": "github",
                                                                          "name": "github",
                                                                          "url": "https://github.com/",
                                                                        },
                                                                        Object {
                                                                          "faIcon": "codepen",
                                                                          "name": "codepen",
                                                                          "url": "https://codepen.io/",
                                                                        },
                                                                      ]
                                                }
                                    />
                        </styled.div>
                        <styled.p>
                                    <strong />
                        </styled.p>
            </div>
            <styled.p />
</styled.div>,
          <ReactTooltip
            insecure={true}
            resizeHide={true}
            wrapper="div"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "imageSource": undefined,
            "isOdd": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <div>
                <styled.div>
                                <styled.h2>
                                                New Project
                                </styled.h2>
                                <ProjectLinksList
                                                links={
                                                                Array [
                                                                                Object {
                                                                                  "faIcon": "github",
                                                                                  "name": "github",
                                                                                  "url": "https://github.com/",
                                                                                },
                                                                                Object {
                                                                                  "faIcon": "codepen",
                                                                                  "name": "codepen",
                                                                                  "url": "https://codepen.io/",
                                                                                },
                                                                              ]
                                                }
                                />
                </styled.div>
                <styled.p>
                                <strong />
                </styled.p>
</div>,
              <styled.p />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <styled.div>
                    <styled.h2>
                                        New Project
                    </styled.h2>
                    <ProjectLinksList
                                        links={
                                                            Array [
                                                                                Object {
                                                                                  "faIcon": "github",
                                                                                  "name": "github",
                                                                                  "url": "https://github.com/",
                                                                                },
                                                                                Object {
                                                                                  "faIcon": "codepen",
                                                                                  "name": "codepen",
                                                                                  "url": "https://codepen.io/",
                                                                                },
                                                                              ]
                                        }
                    />
</styled.div>,
                  <styled.p>
                    <strong />
</styled.p>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <styled.h2>
                        New Project
</styled.h2>,
                      <ProjectLinksList
                        links={
                                                Array [
                                                                        Object {
                                                                          "faIcon": "github",
                                                                          "name": "github",
                                                                          "url": "https://github.com/",
                                                                        },
                                                                        Object {
                                                                          "faIcon": "codepen",
                                                                          "name": "codepen",
                                                                          "url": "https://codepen.io/",
                                                                        },
                                                                      ]
                        }
/>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "class",
                      "props": Object {
                        "children": "New Project",
                      },
                      "ref": null,
                      "rendered": "New Project",
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "function",
                      "props": Object {
                        "links": Array [
                          Object {
                            "faIcon": "github",
                            "name": "github",
                            "url": "https://github.com/",
                          },
                          Object {
                            "faIcon": "codepen",
                            "name": "codepen",
                            "url": "https://codepen.io/",
                          },
                        ],
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": <strong />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "strong",
                  },
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "insecure": true,
            "resizeHide": true,
            "wrapper": "div",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
